### ISSUE TITLE
Add ability to spin up new VPS instances

### ISSUE THREAD

**Issue by iKevinY** [ubclaunchpad/inertia#144]
Some VPS providers like DigitalOcean have a [comprehensive API](https://developers.digitalocean.com/documentation/v2/) that allows programmatic control over server instances. It would be cool if you could just give Inertia your DO API key, and have it handle _everything_, including spinning up new instances!

**Comment by bobheadxi** [comment]
We do want to be careful of building platform-specific features, since it's another level of maintenance, and Inertia's primary aim is cross-platform flexibility with (at the moment) only one dependency (Docker)

Some kind of instance control would definitely be very neat however!

**Comment by bfbachmann** [comment]
I think @bobheadxi makes a good point here. I personally feel that this feature is out of scope for Interia. I think that if a user wants that kind of functionality they should use something like Kubernetes.

**Comment by bobheadxi** [comment]
Yeah, after a bit more thought I think this is outside the scope of what we want Inertia to do. Something similar that I hope Inertia will be able to do is some sort of [Swarm](https://docs.docker.com/engine/swarm/)-like container/cluster orchestration where you can `inertia swarm up` your project on a whole bunch of interconnected VPS's but that's another discussion for a (very) distant day 😅 Closing this ticket for now but thanks for the suggestion!

**Comment by bobheadxi** [comment]
We are looking into provisioning capabilities, starting with Amazon EC2 servers in #261! Reopening this ticket for investigation 💭

**Comment by bobheadxi** [comment]
Renamed this ticket to focus on DigitalOcean

**Comment by bobheadxi** [comment]
Inertia EC2 provisioning PR: https://github.com/ubclaunchpad/inertia/pull/265
Description: https://github.com/ubclaunchpad/strategy/issues/33#issuecomment-453371384

**Comment by bobheadxi** [comment]
https://github.com/digitalocean/godo
